app/services/payment_provider_customers/stripe_service.rb

```diff
@@ -72,6 +73,23 @@ def check_payment_method(payment_method_id)
       result.single_validation_failure!(field: :payment_method_id, error_code: 'value_is_invalid')
     end
 
+    def generate_checkout_url
+      res = Stripe::Checkout::Session.create(
```

Comment:
- We usualy tries to always return a `result` object within the service classes.

What about something like:

```ruby
response = Stripe::Checkout::Session.create(
  # ...

result.url = response['url']
result
```

- Are we sure that the response from Stripe is always a `success`? Meaning do we have to check for some status in the response?

Reply:
About the first point, I do understand however I'm wondering if that would be needed in such a case, as the URL only needs to be used in the webhook and not returned into the result at all. So I'm not 100% we should do it 🤔

About the 2nd, indeed it can be a failure.
I'll check the doc but will probably add a `rescue` to `deliver_error_webhook` as we do in this same file in other methods. I guess it should cover the failure case. There is no failure  status or such we can have in the response body.

Reply:
I've updated the code with
- a result returned, containing a `checkout_url` attribute
- a rescue on the method that uses Stripe client